### PR TITLE
fix(core): allow local module ref with @

### PR DESCRIPTION
**Pull request by grouville**
Fixes https://github.com/dagger/dagger/issues/7762

Current implementation of parseRefString does not allow local refs to have @, as it considers this character as the module version separator

However, this is not supposed to be true for local refs, to which the notion of version is not relevant

### CODE REVIEW

File: core/schema/modulesource.go
Comment (sipsma) [inline core/schema/modulesource.go:243]: nit: the check right below here for `!parsed.hasVersion` will always be `true` now, so it's a bit redundant + confusing. I think with the new logic we can just remove that check now?

**Review by sipsma — APPROVED** [review]
One tiny nit but otherwise LGTM! Can just fix before merging if you agree

File: core/schema/modulesource.go
Comment (grouville) [reply core/schema/modulesource.go:243]: You're right, corrected 🙏

**Comment by grouville** [comment]
Failure is unrelated to PR, merging

**Pull request merged**
